### PR TITLE
sys/shell: Add function to call the shell with a list of strings.

### PR DESCRIPTION
**Pull request by jcarrano**
### Contribution description

The new function makes it easier to call command line tools from within code.

I'm using it to provide a sort of system()/execl() call from within Lua.

### Related PRs

Depends on #10788 to integrate the tests into the main shell tests.

### CODE REVIEW

**Comment by danpetry** [comment]
Re the high level aspects of the PR:
1. Could you please provide tests?
2. Could you please explain briefly why you want to provide a Lua binding to the shell rather than binding to the modules individually, and accessing their commands directly? (Please correct me if my understanding here is inaccurate)

**Comment by jcarrano** [comment]
@danpetry I added the test. The rebase was unintended (our awesome static-test script did it.)

**Comment by jcarrano** [comment]
> Could you please explain briefly why you want to provide a Lua binding to the shell rather than binding to the modules individually, and accessing their commands directly? (Please correct me if my understanding here is inaccurate)

Because I won't write individual bindings to modules (other than as an example). This will allow me to provide a [system()](http://pubs.opengroup.org/onlinepubs/9699919799/functions/system.html)-like (lua has [os.execute](https://www.lua.org/manual/5.3/manual.html#pdf-os.execute) but that takes a single string.) 

The reason for providing such a function from lua is convenience: having an easy to use example application without having to add a lot of code.

**Comment by kaspar030** [comment]
LGTM.

How about merging the test with tests/shell?

File: tests/shell_call/Makefile [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+USEMODULE += shell
+
+DISABLE_MODULE += auto_init
+
+# chronos is missing a getchar implementation
+BOARD_BLACKLIST += chronos
+
+include $(RIOTBASE)/Makefile.include
+
+test:
```
Comment (kaspar030) [inline tests/shell_call/Makefile]: This target is not needed anymore.

Comment (kaspar030) [reply tests/shell_call/Makefile]: anyhow, please merge the test into tests/shell. They're testing the same module, and tests are kinda expensive for the CI...

**Comment by jcarrano** [comment]
I rebased on top of #10788 so that tests can be more easily integrated.

**Comment by jcarrano** [comment]
I'm not rebasing this (yet) again. Closing.

**Pull request closed without merging**